Revamp Volunteer form; move old fields to Register

kfranqueiro opened the pull request:
This PR makes the following changes:

- Moves existing fields from volunteer form into registration form
  - This includes adding validation for those fields and relocating #87
- Adds an extra error region to the registration form near the captcha field, since the form is now likely to span more than a single viewport height
- Implements new fields (checkbox groups) for volunteer form
- Only displays volunteer form when signed in; otherwise, links to sign in page
- Based on further discussion below, removes the timeout from the volunteer form

New breaks intentionally introduced in this PR:

- The checkbox groups in the volunteer form are not described correctly:
  - One uses `role="group"` with `aria-labelledby`, but pointing to the wrong element from a different group
  - One uses `role="group"` with no `aria-labelledby`
  - One does not set `role` at all
  - Discussion item: When we first discussed adding this, it was mentioned that only the first of the three above cases would explicitly fail WCAG 2 (1.3.1 Info and Relationships); should WCAG 3 more clearly cause the remaining two cases to fail?
- The "other" textarea only displays when "other" is checked, with no announcement or label
  - Discussion item: what would be the optimal correct behavior for this?
    - Always displaying the field comes with its own potential snags, e.g. if someone writes in the textbox should it automatically check the "Other" checkbox if it's not checked, and then does that need to be announced anyway, vs. showing/hiding the textarea based on the checkbox state and announcing it?

fstrr commented:
> The "other" textarea only displays when "other" is checked

For me, the `textarea` is visible from the start when the page loads.

kfranqueiro commented:
> For me, the `textarea` is visible from the start when the page loads.

Big oops, thanks for catching. This broke when I switched all of the labels to use `for` instead of containing the `input`s. Should be fixed now.

fstrr commented:
The form times out pretty quickly. If you were going to complete the checkboxes and then write a comment in the textarea, you'd likely not be able to before the timeout happens. We should either lengthen it by a 2-3 minutes or discuss whether we want the timeout on this form as well as the register one.

kfranqueiro commented:
> The form times out pretty quickly. If you were going to complete the checkboxes and then write a comment in the textarea, you'd likely not be able to before the timeout happens. We should either lengthen it by a 2-3 minutes or discuss whether we want the timeout on this form as well as the register one.

If we remove the timeout from the volunteer form, we will still have failures in the gift shop checkout flow's shipping and payment pages. If that's sufficient I'm not against removing the timeout from this page entirely; I don't remember what our motivation was for adding it here.

@rachaelbradley, thoughts?

kfranqueiro committed:
Revamp Volunteer form; move old fields to Register

kfranqueiro commented:
I've rebased this branch and added `@break` JSDoc comments for the failures mentioned in the description.

Based on discussion on Monday RE the comment above, I've removed the timeout from the volunteer form.
